astropy/nddata/nduncertainty.py

```diff
@@ -177,6 +181,14 @@ def parent_nddata(self, value):
             # https://github.com/astropy/astropy/pull/4799#discussion_r61236832
             value = weakref.ref(value)
         self._parent_nddata = value
+        #set uncertainty unit to that of the parent if it was not already set.
+        if self.unit is None:
+            if self.parent_nddata.unit is None:
+                #warning("No units specified")
+                self.unit = None
+            else:
+                #warning("Setting units from that of parent")
+                self.unit = self.parent_nddata
```

Comment:
Doesn't this depend on what kind of uncertainty this is?

Reply:
This is prevented by raising NonImplemented